babyai/model.py

```diff
@@ -211,6 +211,10 @@ def forward(self, obs, memory, instr_embedding=None):
             # outputs: B x L x D
             # memory: B x M
             mask = (obs.instr != 0).float()
+            # quick fix adapted from:      self._get_instr_embedding(obs.instr)
+            if instr_embedding.shape[1] < mask.shape[1]:
+                mask = mask[:, :(instr_embedding.shape[1] - mask.shape[1])]
+
             instr_embedding = instr_embedding[:, :mask.shape[1]]
```

Comment:
I think the easiest fix would be to do

```
mask = mask[:, instr_embedding.shape[1]]
```

instead of 

```
instr_embedding = instr_embedding[:, :mask.shape[1]]
```

I don't really understand why line 218 is currently as it is. The `mask` tensor should be always at least a big as `instr_embedding`. It can be too big though, because `instr_embeddings` might be shorter than `obs.instr` when the last position in a batch is all zeros. 

Locally, this fix works for me. Do you agree that this change is sensible?

It would be also nice to remove the lines from `_get_instr_embedding` that, as you correctly noticed, have no effect whatsoever.

Reply:
`mask = mask[:, instr_embedding.shape[1]]` should be `mask = mask[:, :instr_embedding.shape[1]]` right?
It looks reasonable in my opinion and it shortens the code.

> It would be also nice to remove the lines from _get_instr_embedding that, as you correctly noticed, have no effect whatsoever.

Do you remove them or should i make a second commit with the lines removed?

Reply:
@zynos can you make the commit?

Reply:
@maximecb done

Reply:
Thanks @zynos! Can you please change to using the shorter version that I suggested above, and add the comment to the code that @maximecb suggested below? I think we could then merge this PR.

Alternatively, if you want, I'd be happy to finish fixing this bug myself.

Reply:
@rizar i did that but added a ":" to your suggested line `mask = mask[:, instr_embedding.shape[1]]`
so it is  `mask = mask[:, :instr_embedding.shape[1]]`